src/tools/examples/avtexamples/qtviswinExample.C

```diff
@@ -0,0 +1,333 @@
+VTK_CREATE_CREATE_FUNCTION(vtkVisItRectilinearGrid);
+VTK_CREATE_CREATE_FUNCTION(vtkVisItStructuredGrid);
+
+
+const char*
+vtkVisItObjectFactory::GetVTKSourceVersion()
+{
+    return VTK_SOURCE_VERSION;
+}
+
+vtkVisItObjectFactory::vtkVisItObjectFactory()
+{
+  this->RegisterOverride("vtkCellDataToPointData", "vtkVisItCellDataToPointData",
+                         "vtkVisItCellDataToPointData override vtkCellDataToPointData",
+                         1,
+                         vtkObjectFactoryCreatevtkVisItCellDataToPointData);
+  this->RegisterOverride("vtkRectilinearGrid", "vtkVisItRectilinearGrid",
+                         "vtkVisItRectilinearGrid override vtkRectilinearGrid",
+                         1,
+                         vtkObjectFactoryCreatevtkVisItRectilinearGrid);
+  this->RegisterOverride("vtkStructuredGrid", "vtkVisItStructuredGrid",
+                         "vtkVisItStructuredGrid override vtkStructuredGrid",
+                         1,
+                         vtkObjectFactoryCreatevtkVisItStructuredGrid);
+
+}
+
+//
+// A factory that will allow VisIt to override any vtkObject
+// with a sub-class of that object.
+//
+class vtkVisItGraphicsFactory : public vtkObjectFactory
+{
+  public:
+    static vtkVisItGraphicsFactory * New();
+    vtkTypeMacro(vtkVisItGraphicsFactory, vtkObjectFactory)
+
+    const char* GetDescription() override { return "vtkVisItGraphicsFactory"; };
+    const char* GetVTKSourceVersion() override;
+
+  protected:
+    vtkVisItGraphicsFactory();
+
+  private:
+    vtkVisItGraphicsFactory(const vtkVisItGraphicsFactory&);
+    void operator=(const vtkVisItGraphicsFactory&);
+};
+
+vtkStandardNewMacro(vtkVisItGraphicsFactory)
+
+//
+// Necessary for each object that will override a vtkObject.
+//
+VTK_CREATE_CREATE_FUNCTION(vtkVisItDataSetMapper);
+//VTK_CREATE_CREATE_FUNCTION(vtkVisItRectilinearGrid);
+//VTK_CREATE_CREATE_FUNCTION(vtkVisItStructuredGrid);
+VTK_CREATE_CREATE_FUNCTION(vtkOpenGLPointMapper);
+
+const char*
+vtkVisItGraphicsFactory::GetVTKSourceVersion()
+{
+    return VTK_SOURCE_VERSION;
+}
+
+vtkVisItGraphicsFactory::vtkVisItGraphicsFactory()
+{
+  this->RegisterOverride("vtkDataSetMapper", "vtkVisItDataSetMapper",
+                         "vtkVisItDataSetMapper override vtkDataSetMapper",
+                         1,
+                         vtkObjectFactoryCreatevtkVisItDataSetMapper);
+  this->RegisterOverride("vtkRectilinearGrid", "vtkVisItRectilinearGrid",
+                         "vtkVisItRectilinearGrid override vtkRectilinearGrid",
+                         1,
+                         vtkObjectFactoryCreatevtkVisItRectilinearGrid);
+  this->RegisterOverride("vtkStructuredGrid", "vtkVisItStructuredGrid",
+                         "vtkVisItStructuredGrid override vtkStructuredGrid",
+                         1,
+                         vtkObjectFactoryCreatevtkVisItStructuredGrid);
+  this->RegisterOverride("vtkPointMapper",
```

Comment:
I understand a bit why there are two factories used here, because it follows what happens inside VisIt.  However it seems a bit redundant here in a self-contained example.  Perhaps just use one factory with a comment that it is a combination from two different locations within VisIt? What do you think?

Not related to your PR at all, but I never realized that both factories register overrides for vtkStructuredGrid and vtkRectilinearGrid. That seems odd. AFAICT, any VisIt component that calls InitVTK also calls InitVTKRendering, so it appears we may be creating duplicate overides unnecessarily.

Reply:
I merged the two factories as suggested.